apps/ledger-live-desktop/src/renderer/modals/UpdateFirmwareModal/index.tsx

```diff
@@ -106,17 +107,31 @@ const UpdateModal = ({
   const withFinal = useMemo(() => hasFinalFirmware(firmware?.final), [firmware]);
   const [cancel, setCancel] = useState<boolean>(false);
 
+  const isDisconnectedDeviceError = err instanceof DisconnectedDevice;
+  const isDisconnectedDeviceDuringOperationError = err instanceof DisconnectedDeviceDuringOperation;
+  const isDeviceDisconnected =
+    isDisconnectedDeviceError || isDisconnectedDeviceDuringOperationError;
+
+  console.log("DEVICE", deviceModelId, props.device, updatedDeviceInfo);
+
   const onRequestCancel = useCallback(() => {
-    showDisclaimer || stateStepId === "finish" ? onRequestClose() : setCancel(state => !state);
-  }, [showDisclaimer, stateStepId, onRequestClose]);
+    (showDisclaimer && !isDeviceDisconnected) || stateStepId === "finish" || cancel
+      ? onRequestClose()
+      : setCancel(true);
+  }, [cancel, showDisclaimer, isDeviceDisconnected, stateStepId, onRequestClose]);
 
   const createSteps = useCallback(
     ({ withResetStep }: { withResetStep: boolean }) => {
       const hasRestoreStep =
         firmware && isDeviceLocalizationSupported(firmware.final.name, deviceModelId);
-      const restoreStepLabel = t("manager.modal.steps.restore");
-      const installUpdateLabel =
+
+      const restoreStepLabel =
         stateStepId === "finish"
+          ? t("manager.modal.steps.restoreDone")
+          : t("manager.modal.steps.restore");
+
+      const installUpdateLabel =
+        stateStepId === "finish" || stateStepId === "restore"
```

Comment:
I think that we could improve how we deal with constant values with TS. Could be something like:
```typescript
const STEPS = {
  RESET_DEVICE: "resetDevice",
  ID_CHECK: "idCheck",
  UPDATE_MCU: "updateMCU",
  UPDATING: "updating",
  RESTORE: "restore",
  FINISH: "finish",
} as const;

export type StepId = (typeof STEPS)[keyof typeof STEPS];

...
const installUpdateLabel = stateStepId === STEPS.FINISH ...

```

wdyt?

Reply:
Yes that would be much cleaner, I'll make the modifications